### PR TITLE
Add diagrams to the Istio Reconciler README.md

### PR DESCRIPTION
**Pull request by mmitoraj**
**Description**

Changes made:
- Added diagrams to the Istio Reconciler README.md

See also: https://github.com/kyma-project/kyma/issues/13169
https://github.com/kyma-incubator/reconciler/issues/705

### CODE REVIEW

**Review by werdes72 — APPROVED** [review]

**Review by veichtj — DISMISSED** [review]

File: pkg/reconciler/instances/istio/README.md [hunk 1]
```diff
@@ -4,6 +4,33 @@
...
+The following activity diagram presents detailed steps of the Istio installation/update process.
+
+![Istio Reconciliation](./assets/istio-reconciliation-action.svg)
+
+The reconciliation is executed by the Istio Reconciler. First, it verifies, using the rules explained in the diagram, if the Istio version found on the cluster and the Client version (istioctl) match. If the versions are compatible, either an installation or update process is triggered. Before the update, the version from the Istio [`values.yaml`](https://github.com/kyma-project/kyma/blob/main/resources/istio-configuration/values.yaml) is compared with the cluster version.
+
+If a customer makes changes in the Istio configuration that are not compatible with the Kyma setup configured within `istio-operator.yaml`, the Istio Reconciler automatically overwrites them with the default values.
+
+After choosing the proper Istio version for installation and applying back the default values, the Istio Reconciler patches the Istio Webhook to base on Kyma and Gardener assumptions.
+
```
Comment (klaudiagrz) [inline pkg/reconciler/instances/istio/README.md:33]: I'm wondering if we should include the Architecture section at the very end of the document and treat it as Details? 🤔

File: pkg/reconciler/instances/istio/README.md [hunk 2]
```diff
@@ -4,6 +4,33 @@
...
 
+## Architecture
+
+Reconciliation in Kyma is handled by Reconciler. The Mothership Reconciler knows the reconciliation status of every managed Kyma cluster and initiates reconciliation of all Kyma components.
+
+The reconciliation of Istio is split into two different Kyma components: Istio and Istio Resources. The separation enables having the pure installation process of Istio under control via Istio reconciler implementation and using the Reconciler framework functionality of simply applying charts to the cluster for Istio Resources.
+
+The Istio component installs Istio on a Kyma cluster. For installation purposes, it requires Reconciler that uses `istioctl` and a rendered `istio-operator.yaml` file with Kyma-specific configurations. After proper installation/upgrade of the Istio service mesh within the cluster, the Istio proxies are restarted if needed. Within the whole Kyma reconciliation process, the installation of Istio is a prerequisite as other components rely on proper service mesh setup and on the installed CRDs.
```
[the remark this answers is not in the document]
Comment (mmitoraj) [reply pkg/reconciler/instances/istio/README.md:13]: Lower case I would say - for example: https://istio.io/latest/about/service-mesh/

File: pkg/reconciler/instances/istio/README.md [hunk 3]
```diff
@@ -30,3 +30,33 @@ Follow these steps to run Istio Reconciler on your local machine:
...
+
+The reconciliation of Istio is split into two different Kyma components: Istio and Istio Resources. The separation enables having the pure installation process of Istio under control via Istio Reconciler implementation and the Reconciler framework. The Reconciler framework functionality simply applies charts to the cluster for Istio Resources.
+
+The Istio component installs Istio on a Kyma cluster. For installation purposes, it requires Reconciler that uses `istioctl` and a rendered `istio-operator.yaml` file with Kyma-specific configurations. After proper installation/upgrade of the Istio service mesh within the cluster, the Istio proxies are restarted if needed. Within the whole Kyma reconciliation process, the installation of Istio is a prerequisite as other components rely on the proper service mesh setup and on the installed CRDs.
+
+The Istio Resources component provides charts for additional resources that are related to Istio itself but are not related to the installation process. The resources are as follows:
+
+- Istio monitoring configuration details providing Grafana dashboards specification
+- Istio Ingress Gateway configuring incoming traffic to Kyma
```
Comment (klaudiagrz) [inline pkg/reconciler/instances/istio/README.md:45]:
```suggestion
- Istio Ingress Gateway that configures incoming traffic to Kyma
```

File: pkg/reconciler/instances/istio/README.md [hunk 4]
```diff
@@ -30,3 +30,33 @@ Follow these steps to run Istio Reconciler on your local machine:
...
+- Mutual TLS (mTLS) configuration enabling mTLS cluster-wide in the STRICT mode
```
Comment (klaudiagrz) [inline pkg/reconciler/instances/istio/README.md:46]:
```suggestion
- Mutual TLS (mTLS) configuration that enables mTLS cluster-wide in the STRICT mode
```

File: pkg/reconciler/instances/istio/README.md [hunk 5]
```diff
@@ -30,3 +30,33 @@ Follow these steps to run Istio Reconciler on your local machine:
...
+- Service Monitor configuring monitoring for the Istio component
```
Comment (klaudiagrz) [inline pkg/reconciler/instances/istio/README.md:47]:
```suggestion
- Service Monitor that configures monitoring for the Istio component
```

File: pkg/reconciler/instances/istio/README.md [hunk 6]
```diff
@@ -30,3 +30,33 @@ Follow these steps to run Istio Reconciler on your local machine:
...
+- Istio [VirtualService](https://istio.io/docs/reference/config/networking/virtual-service/) informing whether Istio is up and running
```
Comment (klaudiagrz) [inline pkg/reconciler/instances/istio/README.md:48]:
```suggestion
- Istio [VirtualService](https://istio.io/docs/reference/config/networking/virtual-service/) that informs whether Istio is up and running
```

**Review by klaudiagrz — APPROVED** [review]

**Review by werdes72 — APPROVED** [review]

**Pull request merged**